### PR TITLE
Add scraper for CVS

### LINKED ISSUE #3: add CVS
we want to add CVS to list of scraped availability

### PR DESCRIPTION
**Pull request by zpeyton**
Closes #3

### CODE REVIEW

File: site-scrapers/CVS.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+const siteName = "CVS";
+const site = sites[siteName];
+
+module.exports = async function GetAvailableAppointments(browser) {
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return webData.map((responseLocation) => {
+        let hasAvailability = parseInt(responseLocation.totalAvailable)
```
Comment (johnhawkinson) [inline site-scrapers/CVS.js:11]: Always use a radix with `parseInt`, i.e. `parseInt(responseLocation.totalAvailable, 10)`

File: site-scrapers/CVS.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const page = await browser.newPage();
+    await page.goto(site.website, { waitUntil: "domcontentloaded" });
+    await page.waitForSelector("a[data-modal='vaccineinfo-MA']");
+
+    const [searchResponse, ...rest] = await Promise.all([
+        Promise.race([
+            page.waitForResponse(
+                "https://www.cvs.com/immunizations/covid-19-vaccine.vaccine-status.ma.json?vaccineinfo"
+            ),
+            page.waitForNavigation(),
```
Comment (johnhawkinson) [inline site-scrapers/CVS.js:35]: Err…I am confused why we need to `Promise.race()` and then `waitForNavigation` if we are retrieving 2 pages in sequence.

File: site-scrapers/CVS.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        ]),
+        page.click("a[data-modal='vaccineinfo-MA']"),
+    ]);
+    const response = (await searchResponse.buffer()).toString();
+    responseJSON = JSON.parse(response);
+    rawData = responseJSON.responsePayloadData.data.MA;
```
Comment (johnhawkinson) [inline site-scrapers/CVS.js:41]: We could supplement with actual locations, but I guess that's a 2.0 feature.

File: site-scrapers/CVS.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+async function ScrapeWebsiteData(browser) {
+    let rawData = {};
+    const page = await browser.newPage();
+    await page.goto(site.website, { waitUntil: "domcontentloaded" });
+    await page.waitForSelector("a[data-modal='vaccineinfo-MA']");
+    const [searchResponse, ...rest] = await Promise.all([
+        Promise.race([
+            page.waitForResponse(
+                "https://www.cvs.com/immunizations/covid-19-vaccine.vaccine-status.ma.json?vaccineinfo"
```
Comment (johnhawkinson) [inline site-scrapers/CVS.js:33]: If we're going to start at the toplevel CVS website, we should not hardcode the Mass. JSON page.
(Or we could go directly to the Mass. JSON page).

**Review by johnhawkinson — COMMENTED** [review]
Thanks for fixing the whitespace/comment stuff before I got to it! :)

File: site-scrapers/CVS.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    console.log(`${siteName} starting.`);
+    const webData = await ScrapeWebsiteData(browser);
+    console.log(`${siteName} done.`);
+    return webData.map((responseLocation) => {
+        let hasAvailability = parseInt(responseLocation.totalAvailable)
+            ? true
+            : false;
+        let availability = responseLocation.totalAvailable;
+        return {
+            name: `${siteName} (${responseLocation.city})`,
```
Comment (johnhawkinson) [inline site-scrapers/CVS.js:16]: Is it too much to ask for the city names in Title Case, cf. https://stackoverflow.com/a/4878800?
`      name: 'CVS (NEW BEDFORD)'`
isn't the best.

**Comment by johnhawkinson** [comment]
Can you also return the 
```
    "currentTime": "2021-02-18T19:03:06.091",
```
field in each entry, I guess as `timestamp`, ala #52?

The frontend won't know what to do with it right now, but in the future…

File: site-scrapers/CVS.js (hunk 5 above)
Comment (zpeyton) [reply site-scrapers/CVS.js:17]: Added this code, thanks for the link.

File: site-scrapers/CVS.js (hunk 1 above)
Comment (zpeyton) [reply site-scrapers/CVS.js:11]: Per our convo, it should default to 10. Octal is depreciated.

File: site-scrapers/CVS.js (hunk 2 above)
Comment (zpeyton) [reply site-scrapers/CVS.js]: Good call. Simplified that logic.

File: site-scrapers/CVS.js (hunk 4 above)
Comment (zpeyton) [reply site-scrapers/CVS.js]: Good call. I moved it to the sites obj.

File: site-scrapers/CVS.js (hunk 3 above)
Comment (zpeyton) [reply site-scrapers/CVS.js]: Right. When there are spots available I'll see how difficult it is to get more granular data.

**Review by johnhawkinson — APPROVED** [review]
Looks ready to me.

(I meant something slightly different re the two URLs, but it's not worth fussing over.)

**Review by livgust — APPROVED** [review]

**Pull request merged**